examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1100 @@
sFgsMBhu3xFTU1MDnU6H6OjohzNhIiIiWtCc2vjYq6mpCU1NTYiNjYW/vz+6u7tRUFCAlStXwmQyAQASExMRFhaGl19+Ge+99x5u3LiBN954Azk5OXz7ioiIiAAskO/q8vLyQnl5OTZt2oTVq1cjOzsb4eHhqK+vh06nAwC4u7vj7Nmz8PT0xPr16/HSSy8hNTUV77//vpNnT0RERK5iQd3V5Qj8AEMiIqKFZ0Hc1UVERETkSGx8iIiISDXY+BAREZFqsPEhIiIi1WDjQ0RERKrBxoeIiIhUg40PERERqQYbHyIiIlKNBfGVFY505/Mch4eHnTwTIiIisted39sP+lxmNj73GBkZAQAEBQU5eSZEREQ0VyMjI/Dz85t1nF9ZcY+pqSn09fXB19cXGo3mob3u8PAwgoKCcP36dX4Vhh2Yl/2Ylf2Y1dwwL/sxq7mZj7xEBCMjIzAajXBzm/1KHq743MPNzQ2BgYHz9vqPPfYY/1HMAfOyH7OyH7OaG+ZlP2Y1Nw87r/ut9NzBi5uJiIhINdj4EBERkWq4FxUVFTl7Emrh7u6OjRs3YtEivsNoD+ZlP2ZlP2Y1N8zLfsxqbpyVFy9uJiIiItXgW11ERESkGmx8iIiISDXY+BAREZFqsPEhIiIi1WDj4yClpaUICQmBp6cnoqOj8d133zl7Sk5XVFQEjUZj89Dr9cq4iKCoqAhGoxFeXl7YuHEjOjs7nThjx2loaMALL7wAo9EIjUaDyspKm3F7srFardi5cyeWLFkCb29vvPjii/j1118deRoO86C8tm/fPq3W1q5da7OPGvI6ePAgnn76afj6+mLZsmVITU3FlStXbPZhbd1lT16srduOHj2KiIgI5QMJTSYTvv76a2XcleqKjY8DfPHFF8jPz8e+ffvQ1taG5557DklJSbh27Zqzp+Z0Tz31FCwWi/Lo6OhQxkpKSnDo0CEcPnwYzc3N0Ov1SEhIUL5P7VE2OjqKyMhIHD58eMZxe7LJz89HRUUFysrKcOHCBfz9999ITk7G5OSko07DYR6UFwA8//zzNrX21Vdf2YyrIa/6+nq89tpraGxsRG1tLW7duoXExESMjo4q+7C27rInL4C1BQCBgYEoLi5GS0sLWlpaEB8fj5SUFKW5cam6Epp3zzzzjOTm5tpsCw0NlbfeestJM3INhYWFEhkZOePY1NSU6PV6KS4uVrb9888/4ufnJx999JGjpugSAEhFRYXy3J5shoaGRKvVSllZmbLPb7/9Jm5ubvLNN984bvJOcG9eIiJZWVmSkpIy6zFqzWtgYEAASH19vYiwth7k3rxEWFv34+/vL59++qnL1RVXfObZ+Pg4WltbkZiYaLM9MTERFy9edNKsXEdXVxeMRiNCQkKQmZmJ7u5uAEBPTw/6+/ttctPpdIiLi1N9bvZk09raiomJCZt9jEYjwsPDVZuf2WzGsmXL8OSTTyInJwcDAwPKmFrz+uuvvwAAixcvBsDaepB787qDtWVrcnISZWVlGB0dhclkcrm6YuMzz/744w9MTk4iICDAZntAQAD6+/udNCvX8Oyzz+Kzzz5DdXU1PvnkE/T392PdunUYHBxUsmFu09mTTX9/Pzw8PODv7z/rPmqSlJSEU6dO4fz58/jggw/Q3NyM+Ph4WK1WAOrMS0Swa9cuxMbGIjw8HABr635mygtgbf1XR0cHfHx8oNPpkJubi4qKCoSFhblcXfFztR1Eo9HYPBeRadvUJikpSfl5zZo1MJlMWLlyJU6ePKlcHMjcZvd/slFrflu2bFF+Dg8PR0xMDIKDg3H27FmkpaXNetyjnFdeXh7a29tx4cKFaWOsrelmy4u1ddfq1avx448/YmhoCKdPn0ZWVhbq6+uVcVepK674zLMlS5bA3d19Wsc6MDAwrftVO29vb6xZswZdXV3K3V3MbTp7stHr9RgfH8eff/456z5qZjAYEBwcjK6uLgDqy2vnzp2oqqpCXV0dAgMDle2srZnNltdM1FxbHh4eWLVqFWJiYnDw4EFERkbiww8/dLm6YuMzzzw8PBAdHY3a2lqb7bW1tVi3bp2TZuWarFYrLl++DIPBgJCQEOj1epvcxsfHUV9fr/rc7MkmOjoaWq3WZh+LxYKff/5Z9fkBwODgIK5fvw6DwQBAPXmJCPLy8lBeXo7z588jJCTEZpy1ZetBec1ErbU1ExGB1Wp1vbp6qJdK04zKyspEq9XKsWPH5NKlS5Kfny/e3t7S29vr7Kk51e7du8VsNkt3d7c0NjZKcnKy+Pr6KrkUFxeLn5+flJeXS0dHh2zdulUMBoMMDw87eebzb2RkRNra2qStrU0AyKFDh6StrU2uXr0qIvZlk5ubK4GBgfLtt9/KDz/8IPHx8RIZGSm3bt1y1mnNm/vlNTIyIrt375aLFy9KT0+P1NXViclkkuXLl6sur1dffVX8/PzEbDaLxWJRHmNjY8o+rK27HpQXa+uuPXv2SENDg/T09Eh7e7vs3btX3NzcpKamRkRcq67Y+DjIkSNHJDg4WDw8PCQqKsrmdki12rJlixgMBtFqtWI0GiUtLU06OzuV8ampKSksLBS9Xi86nU42bNggHR0dTpyx49TV1QmAaY+srCwRsS+bmzdvSl5enixevFi8vLwkOTlZrl275oSzmX/3y2tsbEwSExNl6dKlotVqZcWKFZKVlTUtCzXkNVNGAOT48ePKPqytux6UF2vrruzsbOV33NKlS2XTpk1K0yPiWnWlERF5uGtIRERERK6J1/gQERGRarDxISIiItVg40NERESqwcaHiIiIVIONDxEREakGGx8iIiJSDTY+REREpBpsfIiIiEg12PgQERGRarDxIaIFaWBgAK+88gpWrFgBnU4HvV6PzZs34/vvvwcAaDQaVFZWOnmWRORqFjl7AkRE/0d6ejomJiZw8uRJPPHEE/j9999x7tw53Lhxw9lTIyIXxu/qIqIFZ2hoCP7+/jCbzYiLi5s2/vjjj+Pq1avK8+Dg
```

Comment:
> When going beyond 22 qubits, the embedded simulator option shows its limitation in computational power.

I don't think computational power is the bottom line of the difference. SV1 uses c5.24xlarge. If we use c5.24xlarge for embedded sim, it should provides similar computational power. My guesses: Is it because of simulator types (lightning.qubit in embedded sim vs QuEST in SV1)? Or is it because optimization algorithm takes too much memory and competes resource with circuit simulations?

Reply:
The only difference in Job execution time between the embedded simulator and the on-demand simulator is in the different capability of parallelization (batch execute from pennylane). The QNSPSA code is written in a way that the 12 circuits required per optimization step is grouped into two qml.execute() functions, and this is the root cause for the speed-up from the SV1.

Speaking of a single qnode's execution time, lightning.qubit is very similar to QuEST. For a 22-qubit qnode, it takes lightning.qubit 4 sec to execute the circuit (local instance), 6 sec for QuEST (SV1). Considering the overhead of 1 sec from SV1, they are basically the same. Memory is also not the issue for 20+ qubits. The stress test shows that an m5.4xlarge instance provides enough memory the 29 qubit system. And changing from m5.4xlarge to c5.24xlarge changes the execution time by 10%-ish.

Reply:
I think we need to more accurately address the difference of SV1 and embedded sims. As you said, the single node performance is similar. So we should avoid making general statement about which one is better. 

Also, I agree that we don't go into details about the parallelization with SV1, but we can at least mention it and point readers to [doc](https://docs.aws.amazon.com/braket/latest/developerguide/braket-batching-tasks.html?tag=local002-20).